src/main/java/org/candlepin/resource/EnvironmentResource.java

```diff
@@ -300,18 +301,29 @@ public AsyncJobStatusDTO demoteContent(@Verify(Environment.class) String envId,
     @Override
     @SecurityHole(noAuth = true)
     public ConsumerDTO createConsumerInEnvironment(String envId, ConsumerDTO consumer,
-        String userName, String ownerKey, String activationKeys) throws BadRequestException {
+        String userName, String activationKeys) throws BadRequestException {
 
         this.validator.validateConstraints(consumer);
         this.validator.validateCollectionElementsNotNull(consumer::getInstalledProducts,
             consumer::getGuestIds, consumer::getCapabilities);
 
-        //TODO: This needs to support backward compatibility.
-        // Kept as it is to be handled in other task of multi-environment feature.
-        Environment e = lookupEnvironment(envId);
-        consumer.setEnvironments(Arrays.asList(translator.translate(e, EnvironmentDTO.class)));
+        List<EnvironmentDTO> environmentDTOs = Arrays.stream(envId.split("\\s*,\\s*"))
+            .map(this::lookupEnvironment)
+            .map(this.translator.getStreamMapper(Environment.class, EnvironmentDTO.class))
+            .collect(Collectors.toList());
+
+        // Check if all envs belongs to same org
+        boolean sameOwner = environmentDTOs.stream().map(env -> env.getOwner().getKey())
+            .distinct().count() <= 1;
```

Comment:
There shouldn't be a case where count is <0 here. All environments should belong to an owner, so perhaps this should be `== 1`

Reply:
We're also doing a bunch of extra work by using distinct/count.

`.distinct` works internally by building a LinkedHashSet of the parent stream and then streaming its results (with more nuance than that, but that's the gist of it)

`.count` iterates until the stream is exhausted, very similar to a `for (Object obj : objects) ++count;` in cases where the spliterator backing it doesn't provide the exact size.

Given that we only care about the unique value and whether or not we encounter something that isn't equal to the first org we find, we could use a custom reducer to do both at once:

```
BinaryOperator<String> unify = (prev, next) -> {
    if (prev != null && !prev.equals(next)) {
        throw new BadRequestException(i18n.tr("Two or more environments belong to different organizations"));
    }

    return next;
};

String ownerKey = environmentDTOs.stream()
    .map(env -> env.getOwner().getKey())
    .reduce(null, unify);
```